pkg/pillar/zedUpload/ociutil/oci.go

```diff
@@ -121,6 +122,18 @@ func Pull(registry, repo, localFile, username, apiKey string, client *http.Clien
 	}
 	defer w.Close()
 
+	tag, ok := ref.(name.Tag)
+	if !ok {
+		d, ok := ref.(name.Digest)
+		if !ok {
+			err := fmt.Errorf("Image name %s doesn't have a tag or digest", ref)
+			return manifestDirect, manifestResolved, size, err
+		}
+		parts := strings.Split(d.String(), ":")
```

Comment:
No need to play with the string. You can replace all of the `parts := strings.Split(...)` with:

```go
digestTag := fmt.Sprintf("dummyTag-%s", d.DigestStr())
```

Reply:
See https://godoc.org/github.com/google/go-containerregistry/pkg/name#Digest.DigestStr

Reply:
Thanks. I am updating the PR.

Reply:
Resolved.

Reply:
@deitch After these changes, I am seeing new error ```normalize image ref "index.docker.io/library/alpine:dummyTag-sha256:a143f3ba578f79e2c7b3022c488e6e12a35836cd4a6eb9e363d7f3a07d848590": invalid reference format```. It seems like the underlying library is not happy with the format. So, I am reverting these changes.

Reply:
You are correct. I must have been looking at the wrong thing. `DigestStr()` gives `sha256:<hash>`, and not just `<hash>`. You still need to split it on `:`. Sorry.

Reply:
No issues. I updated the PR. Thanks!

Reply:
And there is always a ':' in the d.String()?

Note there is now an unused
const (
	digestTag = "digest-without-tag"
)
to be cleaned up later.

Reply:
> And there is always a ':' in the d.String()?

Yes, d.String() will always return ```<cont-name>@sha256:<hash>```. But it makes sense to check for the length after splitting.

Reply:
It does, but if you use `d.DigestStr()`, it _always_ will return `sha256:<hash>` (which was my mistake in the first place). But still a good idea to check length.